components/tidb_query_datatype/src/codec/collation/collator/utf8mb4_unicode_ci.rs

```diff
@@ -8,41 +8,49 @@ pub struct CollatorUtf8Mb4UnicodeCi;
 
 impl Collator for CollatorUtf8Mb4UnicodeCi {
     type Charset = CharsetUtf8mb4;
+    type Weight = u128;
```

Comment:
Why `CollatorUtf8Mb4Bin`, `CollatorUtf8Mb4GeneralCi` and `CollatorUtf8Mb4UnicodeCi` share the same charset but have different weight?

Reply:
Weight is a sort of "sort key" in terms of collation. Do you have a better alternative name?

Reply:
After reading http://www.unicode.org/reports/tr10/#Weight_Level_Defn , I think `weight` is a good name. 

Maybe we can add a reference here?